src/servers/cron.ts

```diff
@@ -23,7 +23,7 @@ const main = async () => {
     if (result instanceof Error) throw result
   }
 
-  const updatePendingLightningInvoices = () => Wallets.updatePendingInvoices(logger)
+  const updatePendingLightningInvoices = () => Wallets.declineHeldInvoices(logger)
```

Comment:
yes that is correct. I think the better way to frame it would be: 

> This means that we can never have a correct "pending" invoice and all pending should be deleted.

This means that we can have some pending invoices, but in that case they'll be rejected on the next cronjob run. 

We need to make sure we reject them to not have too bad UX for the outside sender (that would result in an issue on our infrastructure) and for the channel to not force closed (maybe lnd auto reject the htlc if there is no explicit rejection now? I know a year+ ago the channel was force closed but I think they changed the behavior)

Reply:
> (maybe lnd auto reject the htlc if there is no explicit rejection now?

Alex still has [this warning line](https://github.com/alexbosworth/lightning/blob/34d95a6ec9033a9d45e46651030f534ca205705b/lnd_methods/invoices/create_hodl_invoice.js#L27) about canceling the invoice before CLTV timeout.

If we get a hung transaction that happens just after the last cronjob, is there a risk that ~24 hrs is more than the CLTV time for someone along the payment path?

Reply:
maybe @alexbosworth can chime in?

Reply:
If you are on lnd 0.14 held invoices will auto cancel themselves

Reply:
do you know when they are cancelled, I guess just before it would lead to force close the channel?

is the time variable (depend on some CLTV settings?) or always the same?

Reply:
yeah it's just before, like within an hour or so, so still you should cancel a held invoice

the time depends on the CLTV since it's relative to the timeout

it only matters if the HTLC is actually held and nothing is happening with it